internal/providers/applehv/applehv.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applehv
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os/exec"
+
+	"github.com/coreos/ignition/v2/config/v3_5_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/vcontext/report"
+	"github.com/mdlayher/vsock"
+)
+
+func init() {
+	platform.Register(platform.Provider{
+		Name:  "applehv",
+		Fetch: fetchConfig,
+	})
+}
+
+func fetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	_, err := f.Logger.LogCmd(exec.Command(distro.ModprobeCmd(), "vsock"), "Loading vsock kernel module")
+	if err != nil {
+		f.Logger.Err("Failed to install vsocker kernel module: %v", err)
+		errors := fmt.Errorf("Failed to install vsock kernel module: %v", err)
+		return types.Config{}, report.Report{}, errors
+	}
+	conn, err := vsock.Dial(2, 1024, &vsock.Config{})
+	if err != nil {
+		return types.Config{}, report.Report{}, err
+	}
+	defer func() {
+		if err := conn.Close(); err != nil {
+			f.Logger.Err("unable to close vsock connection: %v", err)
+		}
+	}()
+
+	req, err := http.NewRequest(http.MethodGet, "http://d/", nil)
```

Comment:
Sorry I am asking for so much but what is this url ?

Reply:
we are using http over vsock; much like unix domain sockets, there still needs to be a "url" of sorts.  this is common practice with restful apis and uds

Reply:
Is the URL here arbitrary / fixed by the server side HTTP server (podman?) or something set by the Apple HV platform?